### ISSUE TITLE
Add k8s_event module to collection

### ISSUE THREAD

**Issue by geerlingguy** [operator-framework/operator-sdk-ansible-util#2]
##### SUMMARY

A `k8s_event` module would be nice to have in certain circumstances. If you want to be able to add events attached to certain resources, this module could help you do that.

See: https://github.com/operator-framework/operator-metering/blob/master/images/metering-ansible-operator/roles/meteringconfig/library/k8s_event.py

##### ISSUE TYPE
- Feature Idea

##### ADDITIONAL INFORMATION

Example task:

```yaml
- name: Create Kubernetes Event
  k8s_event:
    state: present
    name: test-https-emily109
    namespace: default
    message: Event created
    reason: Created
    reportingComponent: Reporting components
    type: Normal
    source:
      component: Metering components
    involvedObject:
      apiVersion: v1
      kind: Service
      name: test-https-emily107
      namespace: default
```

Note: This was migrated from https://github.com/ansible-collections/kubernetes/issues/8

**Comment by tima** [comment]
[This is a larger issue than just this one instance. I'm not even sure which project to file it since it would span the K8s community collection, this one and hopefully more to come.]

We need to be consistent and establish a case convention for ansible+k8s parameters since they use different schemes which is unfortunate and annoying. Take `reportingComponent` or `involvedObject ` in the example above for instance. I'm not 100% sure what the right thing is here. 

My initial reaction is top level keys follow the Ansible convention (snake case) and complex data structures (that are part of the K8s API and probably getting pipe through it) can use the camelCase convention. I'm open to suggestions though.

**Comment by stiller-leser** [comment]
I was about to propose an integration of k8s_event in `community.kubernetes`. But it seems that this is not wished for. I would very much like to see this implemented. A usecase, as compared to `k8s_log`, is for objects like services to be able to react to events like `{"NeutronError": {"message": "Quota exceeded for resources: ['listener'].", "type": "OverQuota", "detail": ""}}` in OpenStack when creating a service with not enough quota.

**Comment by fabianvf** [comment]
Closed with #22
